pkg/porter/credentials.go

```diff
@@ -106,22 +107,91 @@ func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
 	if err != nil {
 		return err
 	}
-	bundle, err := p.CNAB.LoadBundle(opts.CNABFile)
 
+	_, err = p.generateNewCredentialSet(opts.CNABFile, opts.Name, opts.Silent, opts.DryRun)
+	return err
+}
+
+func (p *Porter) chooseCredentialSet(CNABFile string) (string, error) {
+	var credSetNames []string
+	credSets, err := p.Credentials.ReadAll()
 	if err != nil {
-		return err
+		return "", errors.Wrap(err, "failed to read exisiting credential sets")
+	}
+
+	shouldGenerateCred := false
+	var selectedOption string
+	if len(credSets) > 0 {
+		for _, credSet := range credSets {
+			credSetNames = append(credSetNames, credSet.Name)
+		}
+
+		selectCredPrompt := &survey.Select{
+			Message: "Choose a set of credentials to use while installing this bundle",
+			Options: append(credSetNames, generateCredCode, quitGenerateCode),
+			Default: credSetNames[0],
+		}
+		survey.AskOne(selectCredPrompt, &selectedOption, nil)
+
+		switch selectedOption {
+		case generateCredCode:
+			shouldGenerateCred = true
+		case quitGenerateCode:
+			return "", errors.New("Credentials are mandatory to install this bundle")
+		default:
+			return selectedOption, nil
+		}
+	} else {
+		shouldGenerateCredPrompt := &survey.Confirm{
+			Message: "No credential identifier given. Generate one ?",
+		}
+		survey.AskOne(shouldGenerateCredPrompt, &shouldGenerateCred, nil)
+	}
+
+	if !shouldGenerateCred {
+		return "", errors.New("Credentials are mandatory to install this bundle")
 	}
-	name := opts.Name
-	if name == "" {
-		name = bundle.Name
+
+	return p.generateNewCredentialSet(CNABFile, "", false, false)
+}
+
+func (p *Porter) generateNewCredentialSet(CNABFile string, credIdentifierName string, silent bool, dryRun bool) (string, error) {
+	bundle, err := p.CNAB.LoadBundle(CNABFile)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to load bundle while generating credentials")
 	}
+
+	if credIdentifierName == "" {
+		if !silent {
+			inputCredNamePrompt := &survey.Input{
+				Message: "Enter credential identifier name",
+				Default: bundle.Name,
+			}
+			survey.AskOne(inputCredNamePrompt, &credIdentifierName, nil)
+		} else {
+			credIdentifierName = bundle.Name
+		}
+	}
+
 	genOpts := credentialsgenerator.GenerateOptions{
-		Name:        name,
+		Name:        credIdentifierName,
 		Credentials: bundle.Credentials,
-		Silent:      opts.Silent,
+		Silent:      silent,
+		DryRun:      dryRun,
+	}
+
+	fmt.Fprintf(p.Out, "  Generating new credential %s from bundle %s\n", genOpts.Name, bundle.Name)
+	fmt.Fprintf(p.Out, "  %d credentials required for bundle %s\n", len(genOpts.Credentials), bundle.Name)
```

Comment:
How come there is now a two space indent for these lines?

Reply:
Those extra two spaces to align text with survey prompt questions. You know, I work as UI Engineer :)

<img width="262" alt="Screenshot 2020-03-12 at 4 45 09 PM" src="https://user-images.githubusercontent.com/22687681/76516107-de83ea00-6480-11ea-9ef4-1558bb5b4ea8.png">

I will also add comment if you agree to keep those spaces.

Reply:
I apologize, this is super nit-picky. I only asked at first because I really didn't understand where the spaces were coming from.

I believe when the user generates credentials and does specify the name for the credential set, e.g. `porter credentials generate aws` then the extra indent may not be what you desired:

```
$ porter porter credentials generate aws
  Generating new credential aws...
  4 credentials required for bundle mybuns...
? How would you like to set credential 1?
```

To me, the indent only makes sense if it's aligning text that always go with that question. But since they are independent, the alignment is more confusing and disjointed. What do you think?

Reply:
I forgot about `porter credentials generate aws`. That's good catch. Will remove those spaces